### PR TITLE
Added ability to specify NS and SA for K8s pod

### PR DESCRIPTION
**Pull request by yshestakov**
Signed-off-by: Yurii Shestakov <yuriis@nvidia.com>

### CODE REVIEW

File: src/com/mellanox/cicd/Matrix.groovy [hunk 1]
```diff
@@ -750,6 +750,8 @@ def runK8(image, branchName, config, axis, steps=config.steps) {
     def requests = image.requests ?: getConfigVal(config, ['kubernetes', 'requests'], "{memory: 8Gi, cpu: 4000m}")
     def annotations = image.annotations ?: getConfigVal(config, ['kubernetes', 'annotations'], [], false)
     def caps_add = image.caps_add ?: getConfigVal(config, ['kubernetes', 'caps_add'], "[]")
+    def service_account =  getConfigVal(config, ['kubernetes', 'serviceAccount'], "default")
+    def namespace =  getConfigVal(config, ['kubernetes', 'namespace'], "default")
```
Comment (vasily-v-ryabov) [inline src/com/mellanox/cicd/Matrix.groovy:754]: Namespace is also useful in function `build_docker_on_k8`. Not sure about service account.

Comment (yshestakov) [reply src/com/mellanox/cicd/Matrix.groovy:754]: > Namespace is also useful in function `build_docker_on_k8`. Not sure about service account.

no, `docker build` with mounted `/var/run/docker.sock` is executed on the host directly, not under K8s control (accounting.

**Comment by yshestakov** [comment]
bot:retest

**Comment by yshestakov** [comment]
bot:retest

**Comment by yshestakov** [comment]
bot:retest

**Comment by yshestakov** [comment]
bot:retest

**Comment by yshestakov** [comment]
bot:retest

**Comment by mike-dubman** [comment]
very cool! you need to retag "stable" to accomodate that commits
you can take script example for that at ~mdubman/bin/retag

**Pull request merged**